Update AppBar MediaQuery documentation

LongCatIsLooong opened the pull request:
## Description

Explain why you should consider wrapping AppBar in a MediaQuery 

## Related Issues

https://github.com/flutter/flutter/issues/27848

## Tests

I added the following tests:

N/A

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

LongCatIsLooong commented:
@feinstein I made this section less elaborated (than promised), fearing it could be information overload for most readers. Please let me know if find it ambiguous or not helpful!

feinstein commented:
Reading it the first thing that crosses my mind is "why would the ambient `MediaQuery` change?

Maybe the details should be added (and pointed to) at the `MediaQuery` docs as it will be mostly related to it and you could go deeply there.

feinstein commented on packages/flutter/lib/src/material/app_bar.dart:
```diff
@@ -67,17 +67,20 @@ class _ToolbarContainerLayout extends SingleChildLayoutDelegate {
+/// to avoid system UI intrusions. It's taken care of by [Scaffold] when used in
+/// the [Scaffold.appBar] property. When animating an [AppBar], unexpected
+/// [MediaQuery] changes (as is common in [Hero] animations) may cause the content
+/// to suddenly jump. Wrap the [AppBar] in a [MediaQuery] widget with consistent
```
What does `consistent padding values` mean?

LongCatIsLooong replied:
`Wrap the [AppBar] in a [MediaQuery] widget, adjust its padding such that the animation is smooth`?

feinstein replied:
This implies that there isn't a recommended practice, it varies from case to case, is this right?

LongCatIsLooong replied:
Yes that's right. For example for a bottom `AppBar`, the top padding is more likely the culprit of a broken animation.

feinstein replied:
So I guess it's alright as the message conveys it's intent.

feinstein commented:
So I guess it's alright as the message conveys it's intent.

justinmc commented on packages/flutter/lib/src/material/app_bar.dart:
```diff
@@ -67,17 +67,20 @@ class _ToolbarContainerLayout extends SingleChildLayoutDelegate {
+/// to avoid system UI intrusions. It's taken care of by [Scaffold] when used in
+/// the [Scaffold.appBar] property. When animating an [AppBar], unexpected
+/// [MediaQuery] changes (as is common in [Hero] animations) may cause the content
+/// to suddenly jump. Wrap the [AppBar] in a [MediaQuery] widget, adjust its
```
```suggestion
/// to suddenly jump. Wrap the [AppBar] in a [MediaQuery] widget, and adjust its
```

justinmc reviewed (approved):
LGTM 👍

LongCatIsLooong commented:
@googlebot I fixed it.

LongCatIsLooong committed:
update appbar documentation

commit 10d472687194294d559099eff534b0ba03618271
Merge: b990e01d7 35edbe3d9
Author: LongCat is Looong <31859944+LongCatIsLooong@users.noreply.github.com>
Date:   Fri Apr 24 11:39:01 2020 -0700

    Merge remote-tracking branch 'upstream/master' into app-bar-mediaquery-documentation

commit b990e01d70cc6edccb14288c0ba145cea5fdf21a
Author: LongCat is Looong <31859944+LongCatIsLooong@users.noreply.github.com>
Date:   Thu Apr 23 14:47:10 2020 -0700

    review

commit 212c37625ffca34bb7771b90d0b316e687f4529c
Author: LongCat is Looong <31859944+LongCatIsLooong@users.noreply.github.com>
Date:   Fri Feb 21 18:27:09 2020 -0800

    wording

commit 2d71245fe74dcfe99f5f404546c27bab6ca6140b
Author: LongCat is Looong <31859944+LongCatIsLooong@users.noreply.github.com>
Date:   Thu Feb 20 11:13:03 2020 -0800

    Update documentation

commit 4536f926bd12441bbba8834ab99a19a723e63cc6
Author: LongCat is Looong <31859944+LongCatIsLooong@users.noreply.github.com>
Date:   Wed Feb 12 15:07:01 2020 -0800

    App bar documentation
